src/command/setup.rs

```diff
@@ -43,6 +43,19 @@ mod os {
         let home = volta_home()?;
         let formatted_home = format_home(home.root());
 
+        // Don't update the user's shell config files if VOLTA_HOME and PATH already contain what we need.
+        let home_in_path = match env::var_os("PATH") {
+            Some(paths) => env::split_paths(&paths).find(|p| p == home.root()),
+            None => None,
+        };
+
+        if env::var_os("VOLTA_HOME").is_some() && home_in_path.is_some() {
+            debug!(
+                "{} skipping dot-file modification as VOLTA_HOME is set, and included in the PATH."
```

Comment:
It doesn't appear that there's a parameter, so I don't believe we need the `{} ` at the beginning of this (and the formatting will possibly switch with removing it from the string).